### PR TITLE
feat: create useToggle utility hook

### PR DESCRIPTION
**Pull request by MkMan**
A utility hook inspired by this [twitter post](https://twitter.com/kyleshevlin/status/1450259333413892096?s=20) 😃

### CODE REVIEW

File: src/customHooks/useToggle/useToggle.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+    userEvent.click(screen.getByTestId('toggle'));
+    expect(getCurrentState()).toEqual('false');
+
+    userEvent.click(screen.getByTestId('toggle'));
+    expect(getCurrentState()).toEqual('true');
+  });
+
+  it(`should reset the state when reset is called`, () => {
+    render(<TestBed initialValue={true} />);
```
Comment (cyrusxzw) [inline src/customHooks/useToggle/useToggle.spec.tsx:64]: Set initialValue to false is better? which matches the (initialState = false) in useToggle.ts.

Comment (MkMan) [reply src/customHooks/useToggle/useToggle.spec.tsx]: Done 👍

**Review by cyrusxzw — APPROVED** [review]

**Review by hassanraza0054 — APPROVED** [review]
Really love how modular this hook is. Great addition to the gel toolkit @MkMan !

**Comment by andrwchang** [comment]
This really good step forward. Thanks @MkMan great work

**Pull request merged**